### PR TITLE
Add support for enable/disable private vulnerability reporting on repositories

### PR DESCRIPTION
**Pull request by vandanrohatgi**
Closes https://github.com/google/go-github/issues/2883

- [x] go generate
- [x] go vet
- [x] go test

Please let me know if the method names are not descriptive enough or just need to be changed. 

Also, something that's been bugging me. In my previous [PR](https://github.com/google/go-github/pull/2869), I did some actual testing aside from the unit tests. While not present in [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) , is it advisable/necessary/optional to perform actual testing (importing the library, calling the new method and examine response).

### CODE REVIEW

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2077]:
```suggestion
func (s *RepositoriesService) EnablePrivateReporting(ctx context.Context, owner, repo string) (*Response, error) {
```

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2076]:
```suggestion
// GitHub API docs: https://docs.github.com/en/rest/repos/repos#enable-private-vulnerability-reporting-for-a-repository
```

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2096]:
```suggestion
// GitHub API docs: https://docs.github.com/en/rest/repos/repos#disable-private-vulnerability-reporting-for-a-repository
```

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2097]:
```suggestion
func (s *RepositoriesService) DisablePrivateReporting(ctx context.Context, owner, repo string) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @vandanrohatgi !
A few minor tweaks, please, then we will be ready for a second LGTM+Approval before merging.

**Comment by gmlewis** [comment]
> Also, something that's been bugging me. In my previous [PR](https://github.com/google/go-github/pull/2869), I did some actual testing aside from the unit tests. While not present in [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) , is it advisable/necessary/optional to perform actual testing (importing the library, calling the new method and examine response).

In this repo, we do not typically add to the integration tests anymore.
They were extremely helpful in the early design of the repo, but in my opinion, the surface area of this API is so huge that it would be an onerous task to attempt to keep up-to-date in addition to the unit tests.

Having said that, we place a high value on unit tests in this repo and want every endpoint to be exercised "round-trip" with an in-memory http server to verify that every looks like it should work in real practice.

So you are proceeding along just as planned, and we really appreciate your contributions, @vandanrohatgi !

Keep up the great work!  😁 ❤️

**Comment by vandanrohatgi** [comment]
Thank you for making this project such a pleasure to work with!

**Review by gmlewis — APPROVED** [review]
Thank you, @vandanrohatgi !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by mnafees** [comment]
LGTM 👍🏽

**Comment by gmlewis** [comment]
Thank you, @mnafees !
Merging.

**Pull request merged**
